src/UnitOfWork.php

```diff
@@ -3173,10 +3173,14 @@ public function createEntity($className, array $data, &$hints = [])
                     $reflField->setValue($entity, $pColl);
 
                     if ($hints['fetchMode'][$class->name][$field] === ClassMetadata::FETCH_EAGER) {
-                        $isIteration           = isset($hints[Query::HINT_INTERNAL_ITERATION]) && $hints[Query::HINT_INTERNAL_ITERATION];
-                        $isForeignKeyComposite = $targetClass->hasAssociation($assoc['mappedBy']) && count($targetClass->getAssociationMapping($assoc['mappedBy'])['joinColumns'] ?? []) > 1;
-
-                        if ($assoc['type'] === ClassMetadata::ONE_TO_MANY && ! $isIteration && ! $isForeignKeyComposite && ! isset($assoc['indexBy'])) {
+                        if (
+                            $assoc['type'] === ClassMetadata::ONE_TO_MANY
+                            // is iteration
+                            && ! (isset($hints[Query::HINT_INTERNAL_ITERATION]) && $hints[Query::HINT_INTERNAL_ITERATION])
+                            // is foreign key composite
+                            && ! ($targetClass->hasAssociation($assoc['mappedBy']) && count($targetClass->getAssociationMapping($assoc['mappedBy'])['joinColumns'] ?? []) > 1)
+                            && ! isset($assoc['indexBy'])
+                        ) {
```

Comment:
@greg0ire If I'm not mistaken, `mappedBy` should always be present for one-to-many associations.

I also inlined the conditions to avoid checking the same thing twice or making unnecessary method calls. If you think this is unnecessary, I can revert it.

Reply:
makes sense to me 👍